src/CommitmentsV2/SFA.DAS.CommitmentsV2.Api.UnitTests/Controllers/AuthorizationControllerTests.cs

```diff
@@ -112,6 +114,42 @@ public void AuthorizationController_WhenFeatureIsOff_ApprenticeEmailRequired_Sho
 
             Assert.IsInstanceOf<NotFoundResult>(retVal);
         }
+
+        [TestCase(123, 321)]
+        [TestCase(456, 654)]
+        public async Task AuthorizationController_email_optional_sends_correct_query(long employerId, long providerId)
+        {
+            await _fixture.AuthorizationController.OptionalEmail(employerId, providerId);
+
+            _fixture.MediatorMock.Verify(x => x.Send(It.Is<GetEmailOptionalQuery>(q => 
+                q.EmployerId == employerId && q.ProviderId == providerId), CancellationToken.None), Times.Once);
+        }
+
+        [TestCase(123, 321)]
+        [TestCase(456, 0)]
+        [TestCase(0, 987)]
+        public void AuthorizationController_email_optional_test_handler_positive(long employerId, long providerId)
+        {
+            var sut = new GetEmailOptionalQueryHandler(new EmailOptionalConfiguration { EmailOptionalEmployers = new long[] { 123, 456, 789 }, EmailOptionalProviders = new long[] { 321, 654, 987 } });
+            var query = new GetEmailOptionalQuery(employerId, providerId);
+
+            var result = sut.Handle(query, new CancellationToken()).Result;
```

Comment:
I know we can do this, but I personlly hate to see .Result in code, why not use the pattern above and use 'async Task' on the method name